deephyper/core/analytics/_db_manager.py

```diff
@@ -0,0 +1,127 @@
+import json
+import os
+import abc
+import getpass
+import subprocess
+
+import pandas as pd
+import yaml
+from datetime import datetime
+from tinydb import TinyDB, Query
+
+class DBManager(abc.ABC):
+    """Database Manager, for saving DeepHyper experiments and accessing/modifying the resulting databases.
+
+    Example Usage:
+
+        >>> dbm = DBManager(user_name="Bob", path="path/to/db.json")
+
+    Args:
+        user_name (str, optional): the name of the user accessing the database. Defaults to ``os.getlogin()``.
+        path (str, optional): the path to the database. Defaults to ``"~/.deephyper/local_db.json"``.
+    """    
+    def __init__(self, user_name: str = getpass.getuser(), path: str = "~/.deephyper/local_db.json") -> None:
```

Comment:
the previous doc says `os.getlogin()` (update the doc), generally it is better to leave the argument as `user_name: str=None` and then do a check such as
```python
self._user_name = user_name if user_name else getpass.getuser()
```
otherwise the `getpass.getuser()` get executed when the object definition is being executed, which it is preferable to avoid.